gcp-cloud-run/10-network.tf

```diff
@@ -112,9 +118,10 @@ resource "google_project_service" "vpcaccess_api" {
 resource "google_vpc_access_connector" "connector_vault_cloud_run" {
   name     = "${var.deployment_id}-sql-cn"
   provider = google-beta
-  region   = local.vault_region
+  region   = local.pvault_region
+  network = var.create_vpc ? null : var.vpc_id
   subnet {
-    name = module.vpc.subnets["${local.vault_region}/${var.deployment_id}-sb-connector-vault-cloud-run-${var.env}"].name
+    name = var.create_vpc ? module.vpc[0].subnets["${local.pvault_region}/${var.deployment_id}-sb-connector-vault-cloud-run-${var.env}"].name : (var.vault_cn_subnet_id ? var.vault_cn_subnet_id : null)
```

Comment:
Can you move this one and the others that starts with `var.create_vpc` to locals?

```terraform
locals {
	vpc_access_connector_subnet_name = var.create_vpc ? module.vpc[0].subnets["${local.pvault_region}/${var.deployment_id}-sb-connector-vault-cloud-run-${var.env}"].name : (var.vault_cn_subnet_id ? var.vault_cn_subnet_id : null)
}
```